vllm/v1/engine/core.py

```diff
@@ -143,17 +150,36 @@ def abort_requests(self, request_ids: List[str]):
         self.scheduler.finish_requests(request_ids,
                                        RequestStatus.FINISHED_ABORTED)
 
+        self.guided_decoding_manager.remove_requests(request_ids)
+
     def step(self) -> EngineCoreOutputs:
         """Schedule, execute, and make output."""
 
         if not self.scheduler.has_unfinished_requests():
             return EngineCoreOutputs(
                 outputs=[], scheduler_stats=self.scheduler.make_stats())
 
+        # Check for cached grammars and allocate bitmask if necessary
+        self.setup_grammars()
+
         scheduler_output = self.scheduler.schedule()
+
+        if scheduler_output.total_num_scheduled_tokens == 0:
+            return EngineCoreOutputs(
+                outputs=[], scheduler_stats=self.scheduler.make_stats())
+
+        # the bitmask allocation for grammars
+        # should be ready at this point.
```

Comment:
The bitmask allocation is only done once at startup and is not asynchronous (anymore), so this comment isn't relevant anymore.

```suggestion
```